File: src/main/java/net/vanillaoutsider/betterdogs/ai/WildWolfHuntGoal.java
package net.vanillaoutsider.betterdogs.ai;

import net.minecraft.world.entity.LivingEntity;
import net.minecraft.world.entity.ai.goal.target.NearestAttackableTargetGoal;
import net.minecraft.world.entity.ai.targeting.TargetingConditions;
import net.minecraft.world.entity.animal.wolf.Wolf;
import net.vanillaoutsider.betterdogs.config.BetterDogsConfig;

/**
 * Wild wolf hunting behavior.
 * Only hunts prey when health is below threshold.
 */
public class WildWolfHuntGoal<T extends LivingEntity> extends NearestAttackableTargetGoal<T> {

    private final Wolf wolf;

    public WildWolfHuntGoal(Wolf wolf, Class<T> targetType, boolean checkSight,
            TargetingConditions.Selector targetPredicate) {
        super(wolf, targetType, 10, checkSight, false, targetPredicate);
        this.wolf = wolf;
    }

    @Override
    public boolean canUse() {
        // Only applies to wild wolves
        if (wolf.isTame() || wolf.isBaby())
            return false;

        // Only hunt if health is below configurable threshold
        float threshold = BetterDogsConfig.get().wildHuntHealthThreshold;
        if (wolf.getHealth() >= wolf.getMaxHealth() * threshold)
            return false;

        return super.canUse();
    }
}

File: src/main/kotlin/net/vanillaoutsider/betterdogs/ai/WildWolfHuntGoal.kt
package net.vanillaoutsider.betterdogs.ai

import net.minecraft.world.entity.ai.goal.target.NearestAttackableTargetGoal
import net.minecraft.world.entity.animal.wolf.Wolf
import net.minecraft.world.entity.LivingEntity
import net.minecraft.world.entity.ai.targeting.TargetingConditions
import java.util.function.Predicate

/**
 * Wild wolf hunting behavior.
 * Only hunts prey when health is below 50%.
 */
class WildWolfHuntGoal<T : LivingEntity>(
    private val wolf: Wolf,
    targetType: Class<T>,
    checkSight: Boolean,
    targetPredicate: TargetingConditions.Selector? = null
) : NearestAttackableTargetGoal<T>(wolf, targetType, 10, checkSight, false, targetPredicate) {

    override fun canUse(): Boolean {
        // Only applies to wild wolves
        if (wolf.isTame) return false

        // Only hunt if health is below 50%
        if (wolf.health >= wolf.maxHealth * 0.5f) return false

        return super.canUse()
    }
}
